### PR TITLE
[EXPERIMENTAL] OCRexperiment: process all frames in a queue

### PR DESCRIPTION
**Pull request by aquatiko**
Main changes(lines) from the boilerplate is added with a comment(per line) : added/changed by rokus
**Main area of interest(files) are:** [CameraSource.java ](https://github.com/aquatiko/screen-latency-testing/blob/97b07cc4b19149a27b8e85b6db2c90019f5e2c21/ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/ui/camera/CameraSource.java#L1), [OCRCaptureActivity.java](https://github.com/aquatiko/screen-latency-testing/blob/97b07cc4b19149a27b8e85b6db2c90019f5e2c21/ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java#L1), [OCRDetectorProcessor.java](https://github.com/aquatiko/screen-latency-testing/blob/97b07cc4b19149a27b8e85b6db2c90019f5e2c21/ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrDetectorProcessor.java#L1)

## Details about experiment
**Currently the app shows:** FrameID, Frame Capture Time (from app start in ms), Detected Text in frame, Time when result was show on screen(after processing everything) 

**Key points noted**: The average delay between two consecutive frame capture is around ~ 40-50 ms on 60fps. It's due to processing load as each frame is processed in this approach as and when it gets captured.

### CODE REVIEW

File: ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/ui/camera/CameraSource.java [hunk 1]
```diff
@@ -0,0 +1,1226 @@
+/*
+ * Copyright (C) The Android Open Source Project
```
Comment (aquatiko) [inline ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/ui/camera/CameraSource.java:2]: Changes in this file

File: ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrDetectorProcessor.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) The Android Open Source Project
```
Comment (aquatiko) [inline ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrDetectorProcessor.java:2]: Changes in this file

File: ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java [hunk 3]
```diff
@@ -0,0 +1,429 @@
+/*
+ * Copyright (C) The Android Open Source Project
```
Comment (aquatiko) [inline ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java:2]: Changes in this file

**Comment by aquatiko** [comment]
cc @mwarton @patrickmeiring

File: ocr-reader-complete/app/build/intermediates/blame/res/debug/multi-v2/values-ar.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
+{
```
Comment (mwarton) [inline ocr-reader-complete/app/build/intermediates/blame/res/debug/multi-v2/values-ar.json:1]: Please add the build directory and any other generated files to the .gitignore file

File: ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java [hunk 5]
```diff
@@ -0,0 +1,429 @@
...
+
+        gestureDetector = new GestureDetector(this, new CaptureGestureListener());
+        scaleGestureDetector = new ScaleGestureDetector(this, new ScaleListener());
+
+        Snackbar.make(graphicOverlay, "Tap to Speak. Pinch/Stretch to zoom",
+                Snackbar.LENGTH_LONG)
+                .show();
+
+        // Set up the Text To Speech engine.
+        TextToSpeech.OnInitListener listener =
```
Comment (mwarton) [inline ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java:120]: ?  Why is there TTS?

File: ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java [hunk 6]
```diff
@@ -0,0 +1,429 @@
...
+            }
+        }
+
+        // Creates and starts the camera.  Note that this uses a higher resolution in comparison
+        // to other detection examples to enable the text recognizer to detect small pieces of text.
+        cameraSource =
+                new CameraSource.Builder(getApplicationContext(), textRecognizer)
+                .setFacing(CameraSource.CAMERA_FACING_BACK)
+                .setRequestedPreviewSize(200, 200)
+                .setRequestedFps(120.0f)
```
Comment (mwarton) [inline ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java:224]: I think 60 fps should be enough to start

**Comment by patrickmeiring** [comment]
Thanks sending us this PR!
It might be useful (for review purposes) if you separate the OCR sample you started with from the changes you made. So:
Commit 1. Adds the OCR sample you started with (with .gitignore to ignore generated files).
Commit 2. Applies your changes on top.

File: ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java (hunk 5 above)
Comment (aquatiko) [reply ocr-reader-complete/app/src/main/java/com/google/android/gms/samples/vision/ocrreader/OcrCaptureActivity.java:120]: It's another feature of the boilerplate to speak up the text detected on the preview with touch. Haven't looked if that imparts any latency yet.

**Pull request closed without merging**